tests/strategies/settlement/test_settlement_strategy.py

```diff
@@ -81,22 +81,24 @@ def test_event_tick_updates_bids_and_offers(
             strategy_fixture, can_post_settlement_bid, can_post_settlement_offer)
         self.settlement_strategy.event_market_cycle(strategy_fixture)
 
-        strategy_fixture.area.current_tick = 30
         strategy_fixture.area.config = Mock()
         strategy_fixture.area.config.ticks_per_slot = 60
         strategy_fixture.area.config.tick_length = duration(seconds=15)
         self.market_mock.bid.reset_mock()
         self.market_mock.offer.reset_mock()
 
+        strategy_fixture.area.current_tick = 19
+        self.settlement_strategy.event_tick(strategy_fixture)
+        strategy_fixture.area.current_tick = 20
         self.settlement_strategy.event_tick(strategy_fixture)
```

Comment:
Sorry, didn't get the intuition of calling `event_tick ` twice

Reply:
This is related to the change in the SettlementMarketStrategy.event_tick method. You will notice that the order of the offer/bid update and increment/update counter has been reversed (now the offer is updated before the counter is incremented). This is mandatory to have it in that order because AFAIR the  UpdateFrequencyMixin class works that way (Load and PV strategies follow the same logic, first they update their bids and offers, then they increment the counter of the UpdateFrequencyMixin). This has the effect that if the event_tick is only called once, then the bids and the offers of the strategies will not be updated (because the update counter will not have incremented yet). Therefore a second tick is really required in order for the bids and the offers to be updated with a new energy rate.